JSIL/AssemblyTranslator.cs

```diff
@@ -1316,18 +1316,15 @@ Cachers cachers
         }
 
         internal JSFunctionExpression TranslateMethodExpression (
```

Comment:
Passing MethodReference everywhere was artifact of times, when we translated closures inside parent function. We don't need it anymore, so I removed them and changed MethodReference/TypeReference to MethodDefinition/TypeDefinition when possible. It simplifies reading and understanding greatly.